biosimulations/apps/dispatch-service/src/app/app.controller.ts

```diff
@@ -130,15 +140,40 @@ export class AppController {
                     });
                   });
                 });
+              })
+              .on('error', (err) => {
+                console.log('Error occured in file writing', err);
               });
           }
         }
-      })
-
-      
+      });
     }
   }
 
+  @MessagePattern('dispatch_log')
+  async dispatchLog(data: any) {
+    const slurmjobId = data['hpcOutput']['stdout'].match(/\d+/)[0];
+    const simDirSplit = data['simDir'].split('/');
+    const uuid = simDirSplit[simDirSplit.length - 1];
+    // TODO: research more for better duration
+    this.jobMonitorCronJob(slurmjobId, uuid, 30);
+  }
+
+  async jobMonitorCronJob(jobId: string, uuid: string, seconds: number) {
+    const job = new CronJob(`${seconds.toString()} * * * * *`, async () => {
+      const squeueRes = await this.hpcService.squeueStatus(jobId);
+      const jobMatch = squeueRes.stdout.match(/\d+/);
+      const isJobRunning = jobMatch !== null && jobMatch[0] === jobId;
+      if (!isJobRunning) {
+        this.messageClient.emit('dispatch_finish', uuid);
+        this.schedulerRegistry.getCronJob(jobId).stop();
```

Comment:
@bilalshaikh42 That part. I already fixed that 3 commits back, maybe you're seeing the code from previous commits? Can you check if the issue that you're talking about is still there in the latest commits?

Reply:
Yes, this was in reference to the commit [here](https://github.com/biosimulations/Biosimulations/pull/1172#discussion_r488135890)